harness/determined/core/_context.py

```diff
@@ -225,9 +225,8 @@ def init(
 
     # We are on the cluster.
     cert = certs.default_load(info.master_url)
-    session = api.Session(
-        info.master_url, None, None, cert, max_retries=util.get_max_retries_config()
-    )
+    utp = authentication.login_with_cache(info.master_url, cert=cert)
```

Comment:
This looks good. The magic is more explicit after this change. I think it's a good sign for the login part of your patch.

It'd be nice it could be done away with entirely, but that's outside the scope of this refactor.

Also note here that `Session` is a natural choice in this context for a `login` to return.